analyzers/tests/SonarAnalyzer.UnitTest/Rules/DatabasePasswordsShouldBeSecureTest.cs

```diff
@@ -57,6 +61,35 @@ public class DatabasePasswordsShouldBeSecureTest
                                     ParseOptionsHelper.FromCSharp8,
                                     GetReferences("3.1.11", "3.19.80"));
 
+        [DataTestMethod]
+        [DataRow(@"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\Values")]
+        [DataRow(@"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\UnexpectedContent")]
+        [TestCategory("Rule")]
+        public void DatabasePasswordsShouldBeSecure_CS_WebConfig(string root)
+        {
+            var webConfigPath = Path.Combine(root, WebConfig);
+            DiagnosticVerifier.VerifyExternalFile(
+                SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).GetCompilation(),
+                new CS.DatabasePasswordsShouldBeSecure(),
+                File.ReadAllText(webConfigPath),
+                TestHelper.CreateSonarProjectConfig(root, TestHelper.CreateFilesToAnalyze(root, webConfigPath)));
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void DatabasePasswordsShouldBeSecure_CS_CorruptAndNonExistingWebConfigs()
+        {
+            var root = @"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\Corrupt";
+            var nonexisting = @"TestCases\WebConfig\DatabasePasswordsShouldBeSecure\NonExsitingDirectory";
```

Comment:
- I would go for `missing` or `nonExisting`.
- Typo in the directory name `NonExistingDirectory` instead of `NonExsitingDirectory`